drivers/sensor/renesas/rrh46410/rrh46410.c

```diff
@@ -0,0 +1,293 @@
+
+	return 0;
+}
+
+static int rrh46410_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct rrh46410_data *data = dev->data;
+	int rc;
+	uint8_t checksum = ~RRH46410_GET_MEASUREMENT_RESULTS;
+	uint8_t fetch_sample[2];
+	enum sensor_channel_rrh46410 rrh46410_chan = (enum sensor_channel_rrh46410)chan;
+
+	if (chan != SENSOR_CHAN_ALL && rrh46410_chan != SENSOR_CHAN_IAQ &&
+		rrh46410_chan != SENSOR_CHAN_TVOC && rrh46410_chan != SENSOR_CHAN_ETOH &&
+		rrh46410_chan != SENSOR_CHAN_ECO2 && rrh46410_chan != SENSOR_CHAN_RELIAQ) {
+		return -ENOTSUP;
+	}
+
+	fetch_sample[0] = RRH46410_GET_MEASUREMENT_RESULTS;
+	fetch_sample[1] = checksum;
+
+	rc = data->hw_tf->write_data(dev, fetch_sample, sizeof(fetch_sample));
+	if (rc < 0) {
+		LOG_ERR("Failed to send fetch.");
+		return rc;
+	}
+
+	rc = rrh46410_read_sample(dev, &data->sample_counter, &data->iaq_sample,
+	&data->tvoc_sample, &data->etoh_sample, &data->eco2_sample, &data->reliaq_sample);
+	if (rc < 0) {
+		LOG_ERR("Failed to fetch data.");
+		return rc;
+	}
+
+	return 0;
+}
+
+static int rrh46410_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	const struct rrh46410_data *data = dev->data;
+	int32_t convert_val;
+
+	switch ((enum sensor_channel_rrh46410)chan) {
+	case SENSOR_CHAN_IAQ:
+		convert_val = ((int32_t)data->iaq_sample) * 10;
+		val->val1 = convert_val / 100;
+		val->val2 = convert_val % 100;
+		break;
+	case SENSOR_CHAN_TVOC:
+		convert_val = ((int32_t)data->tvoc_sample) * 100;
+		val->val1 = convert_val / 1000000;
+		val->val2 = convert_val % 1000000;
+		break;
+	case SENSOR_CHAN_ETOH:
+		convert_val = ((int32_t)data->etoh_sample) * 100;
+		val->val1 = convert_val / 10000;
+		val->val2 = convert_val % 10000;
+		break;
+	case SENSOR_CHAN_ECO2:
+		convert_val = (int32_t)data->eco2_sample;
+		val->val1 = convert_val / 100;
+		val->val2 = convert_val % 100;
+		break;
+	case SENSOR_CHAN_RELIAQ:
+		convert_val = ((int32_t)data->reliaq_sample) / 10;
+		val->val1 = convert_val * 10;
+		val->val2 = convert_val % 10;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int rrh46410_init(const struct device *dev)
+{
+	const struct rrh46410_config *cfg = dev->config;
+	int err;
+	int status;
+	uint8_t rrh46410_mode;
+
+	LOG_DBG("Initializing %s", dev->name);
+
+	status = cfg->bus_init(dev);
+
+	if (status < 0) {
+		return status;
+	}
+
+	if (!gpio_is_ready_dt(&cfg->reset_gpio)) {
+		LOG_ERR("The reset pin GPIO port is not ready.\n");
+		return 0;
+	}
+
+	err = gpio_pin_configure_dt(&cfg->reset_gpio, GPIO_OUTPUT_INACTIVE);
+	if (err != 0) {
+		LOG_ERR("Configuring GPIO pin failed: %d\n", err);
+		return 0;
+	}
+
+	err = gpio_pin_set_dt(&cfg->reset_gpio, 1);
+	if (err != 0) {
+		LOG_ERR("Setting GPIO pin level failed: %d\n", err);
+	}
+
+	k_sleep(K_MSEC(100));
+
+	err = gpio_pin_set_dt(&cfg->reset_gpio, 0);
+	if (err != 0) {
+		LOG_ERR("Setting GPIO pin level failed: %d\n", err);
+	}
+
+	k_sleep(K_MSEC(600));
```

Comment:
Are you sure you want to add 700 ms of boot delay?

Reply:
@MaureenHelm @teburd there are a lot of sensors like this. I'm starting to think that we should have some pattern for a 2+ phase init for sensor drivers. We can kinda fake it right now. We have https://docs.zephyrproject.org/latest/kernel/drivers/index.html#deferred-initialization so we could call `device_init()` on a `SYS_INIT` which would kick off the process, then instead of sleeping we could use a timer or a deferred work item to call `device_init()` again and continue the init process until it finishes. The device would have to maintain an internal state (maybe a place to use the fsm subsystem)?

Reply:
The large delays here are concerning, we don't have a great story on how to deal with it today as init is not based on asynchronous state machines, its based on a blocking init call done from the init loop done per level here https://github.com/zephyrproject-rtos/zephyr/blob/main/kernel/init.c#L347

Meaning this will block your application from starting for nearly a second and that would likely be surprising to people.